src/Control/Director.php

```diff
@@ -1046,4 +1008,20 @@ public static function get_template_global_variables()
             'BaseHref' => 'absoluteBaseURL',    //@deprecated 3.0
         );
     }
+
+    /**
+     * Helper to validate or check the current request object
+     *
+     * @param HTTPRequest $request
+     * @return HTTPRequest Request object if one is both current and valid
+     */
+    protected static function currentRequest(HTTPRequest $request = null)
+    {
+        // Ensure we only use a registered HTTPRequest and don't
+        // incidentally construct a singleton
+        if (!$request && Injector::inst()->has(HTTPRequest::class)) {
```

Comment:
When we've removed the registered instance. It's in the tests (see new InjecorTest.php).